logger/set/all/logger.rb

```diff
@@ -0,0 +1,103 @@
+require_dependency 'logger'
+
+event :start_performance_logger_on_change, :before=>:handle, :when => proc { |c| c.performance_log? } do
+  start_performance_logger
+  @handle_logger = true
+end
+event :stop_performance_logger_on_change, :after=>:handle, :when => proc { |c| c.performance_log? } do
+  stop_performance_logger
+  @handle_logger = false
+end
+
+event :start_performance_logger_on_read, :before=>:show, :when => proc { |c| c.performance_log? }, :on=>:read do
+  start_performance_logger unless @handle_logger
+end
+event :stop_performance_logger_on_read, :after=>:show, :when => proc { |c| c.performance_log? }, :on=>:read do
+  stop_performance_logger unless @handle_logger
+end
+
+
+event :request_logger, :after=>:show, :when => proc { |c| Card.config.request_logger } do
+  ::Logger::Request.write_log_entry Env[:controller]
+end
+
+def start_performance_logger
+  if Env.params[:performance_log]
+    ::Logger::Performance.load_config Env.params[:performance_log]
+  end
+
+  method = Env[:controller].env["REQUEST_METHOD"]
+  path   = Env[:controller].env["PATH_INFO"]
+  ::Logger::Performance.start :method=>method, :message=>path, :category=>'format'
+end
+
+def stop_performance_logger
+  ::Logger::Performance.stop
+  if Env.params[:perfomance_log]
+     ::Logger::Performance.load_config( Card.config.performance_logger || {} )
+  end
+end
+
+def performance_log?
+  Env.params[:performance_log] || Card.config.performance_logger
+end
+
+
+module ClassMethods
+  def execute_query query
+    ::Logger.with_logging :search, :message=>query.query, :details=>query.sql.strip do
+      Card::Set::All::Collection::ClassMethods.instance_method(:execute_query).bind(binding).call(query)
+    end
```

Comment:
This one was tough. It's a singleton method on Card defined in a set module so I couldn't use `alias_method` here to get the original method.